docs/architecture/CLI-TEST-ARCHITECTURE.md

```
@@ -0,0 +1,502 @@
+graph TD
+    A[Test Case] --> B[Write .tsp file to disk]
+    B --> C[Run: npx tsp compile]
+    C --> D[Read output from disk]
+    D --> E[Parse AsyncAPI YAML/JSON]
+    E --> F[Run assertions]
+    F --> G[Cleanup temp files]
+```
+
+### Directory Structure
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)